Client/Frontend/Library/HistoryPanel/HistoryPanel.swift

```diff
@@ -0,0 +1,651 @@
+                cell.leftImageView.tintColor = .theme.browser.tint
+                cell.leftImageView.backgroundColor = .theme.homePanel.historyHeaderIconsBackground
+                cell.accessibilityIdentifier = historyActionable.itemA11yId
+                self.setTappableStateAndStyle(with: historyActionable, on: cell)
+                
+                return cell
+            }
+            
+            if let site = item as? Site {
+                guard let cell = tableView.dequeueReusableCell(withIdentifier: TwoLineImageOverlayCell.reuseIdentifier, for: indexPath) as? TwoLineImageOverlayCell else {
+                    self.browserLog.error("History Panel - cannot create TwoLineImageOverlayCell for site!")
+                    return nil
+                }
+                
+                cell.titleLabel.text = site.title
+                cell.titleLabel.isHidden = site.title.isEmpty
+                cell.descriptionLabel.text = site.url
+                cell.descriptionLabel.isHidden = false
+                cell.leftImageView.layer.borderColor = HistoryPanelUX.IconBorderColor.cgColor
+                cell.leftImageView.layer.borderWidth = HistoryPanelUX.IconBorderWidth
+                cell.leftImageView.contentMode = .center
+                cell.leftImageView.setImageAndBackground(forIcon: site.icon, website: site.tileURL) { [weak cell] in
+                    cell?.leftImageView.image = cell?.leftImageView.image?.createScaled(CGSize(width: HistoryPanelUX.IconSize, height: HistoryPanelUX.IconSize))
+                }
+                
+                return cell
+            }
+            
+            if let searchTermGroup = item as? ASGroup<Site> {
+                guard let cell = tableView.dequeueReusableCell(withIdentifier: TwoLineImageOverlayCell.accessoryUsageReuseIdentifier, for: indexPath) as? TwoLineImageOverlayCell else {
+                    self.browserLog.error("History Panel - cannot create TwoLineImageOverlayCell for STG!")
+                    return nil
+                }
+                
+                if let groupCount = searchTermGroup.description {
+                    cell.descriptionLabel.text = "\(groupCount) sites"
+                }
+                
+                cell.titleLabel.text = searchTermGroup.displayTitle
+                cell.leftImageView.layer.borderWidth = 0
+                cell.leftImageView.contentMode = .center
+                cell.chevronAccessoryView.isHidden = false
+                cell.leftImageView.setImageAndBackground(forIcon: nil, website: nil) { [weak cell] in
+                    cell?.leftImageView.image = cell?.leftImageView.image?.createScaled(CGSize(width: HistoryPanelUX.IconSize, height: HistoryPanelUX.IconSize))
+                    cell?.leftImageView.image = UIImage(named: "recently_closed")
+                }
+                
+                return cell
+            }
+            
+            // This should never happen! You will have an empty row!
```

Comment:
```suggestion
            // This should never happen as it would be an empty row.
```

Reply:
Haha I kinda like the original b/c it gives this sense of PANIC out.